devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy

```diff
@@ -1176,3 +1197,72 @@ private boolean isMasterCode(codeID) {
 	}
 	(codeID == 0) ? true : false
 }
+
+/**
+ * API end-point for updating the 'lockCodesData' attribute with the user info
+ *
+ * @param userInfo: Map consisting values for code id, phone number, access type, and last usage
+ */
+def setUserInfo(userInfo) {
+	log.trace "[DTH] Executing 'setUserInfo()' by ${device.displayName} with userInfo := ${userInfo}"
+	def codeID = userInfo.codeID
+	if (codeID) {
+		codeID = codeID.toString()
+		def lockCodesData = loadlockCodesData(codeID)
+		lockCodesData[codeID].lastUsage = userInfo.lastUsage
+		lockCodesData[codeID].userInfo.phoneNum = userInfo.phoneNum ?: ""
```

Comment:
What is phoneNum used for? This doesn't seem like the right place to be storing phone numbers.

Reply:
In DTH we store code name for e.g. "Plumber" etc., in lockCodes attribute, which is actually the guest name. This was implemented in Smart Locks V1. The reason was that if user uninstalls the smartapp and installs it again then lockCodes data shouldn't get erased and hence it was stored in DTH.

In Smart Locks V2 this is extended to storing guest's phone number also. If we shift the phone number to SmartApp then the solution will break if user uninstalls and installs the SmartApp. In this case the code name will be picked up from DTH but phone number will be missing.